LiveSched/src/main/java/dev/coms4156/project/livesched/MyFileDatabase.java

```diff
@@ -129,6 +199,34 @@ public void saveContentsToFile(int contentType) {
     }
   }
 
+  /**
+   * Uploads a local file to Google Cloud Storage.
+   *
+   * @param bucketName the name of the GCS bucket
+   * @param objectName the name to save the object as in GCS
+   * @param filePath   the path of the local file to upload
+   * @throws IOException if the upload fails
+   */
+  private void uploadFileToCloud(String bucketName, String objectName, String filePath)
+      throws IOException {
+    BlobId blobId = BlobId.of(bucketName, objectName);
+    BlobInfo blobInfo = BlobInfo.newBuilder(blobId).build();
+
+    // Add a precondition to avoid race conditions
+    Storage.BlobWriteOption precondition;
+    if (storage.get(bucketName, objectName) == null) {
+      precondition = Storage.BlobWriteOption.doesNotExist();
+    } else {
+      precondition = Storage.BlobWriteOption.generationMatch(
+          storage.get(bucketName, objectName).getGeneration());
+    }
+
+    storage.createFrom(blobInfo, Paths.get(filePath), precondition);
```

Comment:
Just wondering: Should we try to overwrite the content if content exists in given bucket?

Reply:
After a couple of tests, I noticed that the default is to overwrite the file if we try to upload a file with the same name. So, I think it should be good!

Reply:
Perfect.